test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(JSON.stringify(requestBody));
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = JSON.stringify({ foo: 'bar' });
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(requestBody);
+      });
+    });
+
+    describe('Submitting a PUT request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(204, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+    });
+
+    describe.skip('Submitting a PATCH request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(206, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.patch({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(206);
+        expect(body).toBeFalsy();
```

Comment:
same note as other (similar) comment: this test doesn't seem to be enforcing anything useful about `body` ?